streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -655,7 +655,11 @@ void suspend(final boolean clean,
             }
         } else {
             // In the case of unclean close we still need to make sure all the stores are flushed before closing any
-            super.flushState();
+            try {
+                stateMgr.flush();
```

Comment:
I'm wondering which commits in trunk that are not in 2.5 that fixed this and below issues? Could we just cherry-pick that commit to 2.5 instead?